pennylane/devices/qubit/apply_operation.py

```diff
@@ -54,7 +56,7 @@ def _get_slice(index, axis, num_axes):
     return tuple(idx)
 
 
-def apply_operation_einsum(op: qml.operation.Operator, state):
+def apply_operation_einsum(op: qml.operation.Operator, state, batch_dim=0):
```

Comment:
can you add batch_dim to the docstring args (and all other docstrings that need updating)? I'm mostly asking bc if I'm following, I'm pretty sure it's always either 0 or 1 - in other words, a boolean! your tests would agree.

Reply:
Thanks for the reminder!

Reply:
Sorry, I should have been clearer in that comment. If `batch_dim` is indeed always one of `{0, 1}`, it should actually _be_ a boolean, and I'd prefer you update it accordingly

Reply:
This parameter is essentially just saying "This is the offset between the wire number and the dimension it needs to be applied to". I wonder if we can come up with a name that indicates that.  Like even `batching_offset`?

Reply:
my preferred interface is to make the parameter a bool called something like `is_batched`. Then in the function, say `batch_offset = int(is_batched)` and use that. To confirm:
```pycon
>>> int(True), int(False)
(1, 0)
```

Reply:
(The casting is done internally if you just want addition: 
```python
>>> 4 + True
5
```

Reply:
So how about `state_is_batched: bool =False`.

Reply:
I think I agree with @timmysilv 's naming suggestion, because the information passed to `apply_operation_einsum` is whether the state is batched or not. Only within the function is this translated to "shift some axis by the corresponding offset".
I think we should include `state` in the variable name, because both `op` and `state` could be batched, but the variable only is used to pass information on the `state`.